fe/fe-core/src/main/java/org/apache/doris/datasource/mapping/DefaultIdentifierMapping.java

```diff
@@ -0,0 +1,268 @@
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class DefaultIdentifierMapping implements IdentifierMapping {
+    private static final Logger LOG = LogManager.getLogger(DefaultIdentifierMapping.class);
+
+    private final ObjectMapper mapper = new ObjectMapper();
+    private final ConcurrentHashMap<String, String> localDBToRemoteDB = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, ConcurrentHashMap<String, String>> localTableToRemoteTable
+            = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, ConcurrentHashMap<String, ConcurrentHashMap<String, String>>>
+            localColumnToRemoteColumn = new ConcurrentHashMap<>();
+
+    private final boolean isLowerCaseMetaNames;
+    private final String metaNamesMapping;
+
+    public DefaultIdentifierMapping(boolean isLowerCaseMetaNames, String metaNamesMapping) {
+        this.isLowerCaseMetaNames = isLowerCaseMetaNames;
+        this.metaNamesMapping = metaNamesMapping;
+    }
+
+    private boolean isMappingInvalid() {
+        return metaNamesMapping == null || metaNamesMapping.isEmpty();
+    }
+
+    @Override
+    public List<String> fromRemoteDatabaseName(List<String> remoteDatabaseNames) {
+        // If mapping is not required, return the original input
+        if (!isLowerCaseMetaNames && isMappingInvalid()) {
+            return remoteDatabaseNames;
+        }
+        JsonNode databasesNode = readAndParseJson(metaNamesMapping, "databases");
+
+        Map<String, String> databaseNameMapping = Maps.newTreeMap();
+        if (databasesNode.isArray()) {
+            for (JsonNode node : databasesNode) {
+                String remoteDatabase = node.path("remoteDatabase").asText();
+                String mapping = node.path("mapping").asText();
+                databaseNameMapping.put(remoteDatabase, mapping);
+            }
+        }
+
+        Map<String, List<String>> result = nameListToMapping(remoteDatabaseNames, localDBToRemoteDB,
+                databaseNameMapping, isLowerCaseMetaNames);
+        List<String> localDatabaseNames = result.get("localNames");
+        List<String> conflictNames = result.get("conflictNames");
+        if (!conflictNames.isEmpty()) {
+            throw new RuntimeException(
+                    "Conflict database/schema names found when lower_case_meta_names is true: " + conflictNames
+                            + ". Please set lower_case_meta_names to false or"
+                            + " use meta_name_mapping to specify the names.");
+        }
+        return localDatabaseNames;
+    }
+
+    @Override
+    public List<String> fromRemoteTableName(String remoteDbName, List<String> remoteTableNames) {
+        // If mapping is not required, return the original input
+        if (!isLowerCaseMetaNames && isMappingInvalid()) {
+            return remoteTableNames;
```

Comment:
If user set `Config.lower_case_table_names = 1`,
I think we also need to handle the name mapping?

Reply:
In IdentifierMapping, we only care about the lower_case_meta_names attribute in catalog, not the lower_case_table_names in fe.conf. We put fe.conf in ExternalDatabase for processing.